stdlib/3/sys.pyi

```diff
@@ -214,3 +214,12 @@ if sys.version_info >= (3, 8):
     unraisablehook: Callable[[UnraisableHookArgs], Any]
     def addaudithook(hook: Callable[[str, Tuple[Any, ...]], Any]) -> None: ...
     def audit(__event: str, *args: Any) -> None: ...
+
+if sys.version_info >= (3, 6):
+    from typing import AsyncGenerator
+    _AsyncgenHook = Callable[[AsyncGenerator[Any, Any]], None]
```

Comment:
This should be Optional

Reply:
Do you mean the whole type, because it’s permitted to not provide one of the hooks if you want? E.g.
```python
_AsyncgenHook = Optional[Callable[[AsyncGenerator[Any, Any]], None]]
```

Do you prefer this over putting the optional status in the individual function parameters/returns?

Reply:
Yeah, I meant the whole type. It doesn't just affect `set_asyncgen_hooks`, the fields of `_asyncgen_hooks` can also be None. Since that's everywhere the alias is used, we might as well change the alias itself.

Reply:
Right, it looks a bit odd to me but it would certainly work fine. 455e2085a37e